File: Python/MergeSort.py
def Merge(a,b,arr):
    k=0
    i=0
    j=0

    while((i<len(a))and(j<len(b))):
        if(a[i]<b[j]):
            arr[k]=a[i]
            i+=1
            k+=1
        else:
            arr[k]=b[j]
            j+=1
            k+=1



    if(i<len(a)):
        for z in range(i,len(a)):
            arr[k]=a[z]
            k+=1

    else:
        for z in range(j,len(b)):
            arr[k]=b[z]
            k+=1

    return


def Mergesort(arr):
    if(len(arr)==1):
        return
    else:
        m=len(arr)//2
        n=len(arr)-m

        a=[]
        b=[]
        k=0
        for i in range(0,m):
            a.append(arr[i])
            k+=1

        for i in range(k, len(arr)):
            b.append(arr[i])

    Mergesort(a)
    Mergesort(b)

    Merge(a,b,arr)

    return



mylist=[]
j=100000
for i in range(100000):
    mylist.append(j)
    j-=1

Mergesort(mylist)

print(mylist)

File: Javascript/MergeSort.js
/*This program implements the merge sort algorithm which is a stable sort but not in-place */

function merge(arr,start,mid,end)
{
    arr1 = [];
    arr2 = [];

    var i = start;
    var j = mid+1;

    while(i<=mid)
    {
        arr1.push(arr[i++]);
    }
    while(j<=end)
    {
        arr2.push(arr[j++]);
    }

    i = j = 0;
    var k = start;
    while((i<arr1.length)&&(j<arr2.length))
    {
        if(arr1[i]<arr2[j])
        {
            arr[k++] = arr1[i++];
        }
        else
        {
            arr[k++] = arr2[j++];
        }
    }

    while(i<arr1.length)
    {
        arr[k++] = arr1[i++];
    }
    while(j<arr2.length)
    {
        arr[k++] = arr2[j++];
    }
}

function sort(arr,start,end)
{
    if(start<end)
    {
        var mid = Math.floor((start+end)/2);
        sort(arr,start,mid);
        sort(arr,mid+1,end);
        merge(arr,start,mid,end);
    }
}

var arr = [];
for(var i=100,j=0;i>0;i--)
{
    arr[j++]=i;
}
console.log(arr);

sort(arr,0,arr.length-1);
console.log(arr);